### PR TITLE
feat(opensearchservice): cold storage option

### LINKED ISSUE #29366: Amazon OpenSearch Service: High Level Constructs for ColdStorage Options
### Describe the feature

The OpenSearch Team allows customers to configure Cold Storage options on their cluster which is a cost efficient way to store infrequently and historical data  https://docs.aws.amazon.com/opensearch-service/latest/developerguide/cold-storage.html

The CFN support for this feature is already there please refer: [CFN Docs link for ColdStroageOptions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-opensearchservice-domain-coldstorageoptions.html)

### Use Case

he general recommendation is to use high level constructs and due to lack of High level construct for this feature, the CDK template needs to be migrated to use CFN constructs if there is a need to Create an OpenSearchDomain with ColdStorageOptions. This serves as a hinderance to the adoption of the ColdStorageOptions feature for AWS OpenSearch Domains.

### Proposed Solution

Add a seperate field for ColdStorageOptions in the existing [Domain](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_opensearchservice.Domain.html) CDK construct refer below
const domain = new Domain(this, 'Domain', {
  version: EngineVersion.OPENSEARCH_1_0,
  capacity: {
    masterNodes: 2,
    warmNodes: 2,
    warmInstanceType: 'ultrawarm1.medium.search',
  },
 **ColdStorageOptions: {
 Enabled: true
}**
});

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.84.0

### Environment details (OS name and version, etc.)

macOS Ventura 13.4

### PR DESCRIPTION
**Pull request by msambol**
Closes #29366. 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts [hunk 1]
```diff
@@ -687,6 +687,15 @@ export interface DomainProps {
    * @default - false
    */
   readonly suppressLogsResourcePolicy?: boolean;
+
+  /**
+   * Whether to enable or disable cold storage on the domain. You must enable UltraWarm storage to enable cold storage.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-opensearchservice-domain-coldstorageoptions.html
+   *
+   * @default - undefined
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:696]: defaulted this to `undefined` so there isn't changes in already deployed templates, but could also make this `false` if we want.

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts [hunk 2]
```diff
@@ -687,6 +687,15 @@ export interface DomainProps {
    * @default - false
    */
   readonly suppressLogsResourcePolicy?: boolean;
+
+  /**
+   * Whether to enable or disable cold storage on the domain. You must enable UltraWarm storage to enable cold storage.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-opensearchservice-domain-coldstorageoptions.html
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:694]:
```suggestion
   * @see https://docs.aws.amazon.com/opensearch-service/latest/developerguide/cold-storage.html
```
Let's provide the link to the dev guide instead.

**Review by lpizzinidev — APPROVED** [review]
Looks good, thanks 👍
I would just update a link in the documentation.

**Review by aaythapa — APPROVED** [review]
Thank you for the PR! LGTM

**Pull request merged**